### PR TITLE
Remove usage of globals in collector package

### PR DESCRIPTION
**Pull request by rfratto**
This PR creates a new `collector.Config` interface to represent an uninitialized collector. It is modeled after [Prometheus' discover.Config](https://github.com/prometheus/prometheus/blob/9486be535bbfbff65d858f94720b5e467caead4d/discovery/discovery.go#L47) system. 

The main package has been updated to use these interfaces instead of globals. 

This PR is one of a few to slowly move collectors over to support this new interface. For this initial PR, all collectors have been temporarily disabled. 

As an additional change, the `main.windowsCollector` struct has been moved to `collector.Prometheus` so it may be used by downstream by importers (eg the Grafana Agent).

### CODE REVIEW

File: collector/collector.go [hunk 1]
```diff
@@ -50,52 +50,30 @@ func getWindowsVersion() float64 {
...
 
-func registerCollector(name string, builder collectorBuilder, perfCounterNames ...string) {
-	builders[name] = builder
-	addPerfCounterDependencies(name, perfCounterNames)
-}
+	// RegisterFlags should add config options to the provided Kingpin
+	// application.
+	RegisterFlags(ka *kingpin.Application)
 
-func addPerfCounterDependencies(name string, perfCounterNames []string) {
```
Comment (rfratto) [inline collector/collector.go:65]: Mapping counters got lost and needs to be reimplemented somewhere.

File: collector/collector.go [hunk 2]
```diff
@@ -50,52 +50,30 @@ func getWindowsVersion() float64 {
...
+type Config interface {
+	// Name should return the name of the collector.
+	Name() string
 
-var (
-	builders                = make(map[string]collectorBuilder)
-	perfCounterDependencies = make(map[string]string)
-)
+	// PerfCounterNames should return perf counter keys used by the collector.
+	PerfCounterNames() []string
```
Comment (rfratto) [inline collector/collector.go]: Maybe this should be attached to the collector and not the config? The only time this is called is when the only available object is the collector.

File: collector/collector.go (hunk 1 above)
Comment (rfratto) [reply collector/collector.go:65]: It can be done in getPerfQuery in a very unoptimized form and use memoization later.

**Comment by mattdurham** [comment]
~~Is this PR expected to be a stepping stone and not functional yet?~~ Totally skipped over that they are disabled.

**Review by mattdurham — APPROVED** [review]
Looks legit for step 1

**Pull request merged**